src/credential.rs

```diff
@@ -117,6 +131,7 @@ pub trait ProvideAwsCredentials {
 }
 
 /// Provides AWS credentials from environment variables.
+#[derive(Debug, Clone)]
```

Comment:
ah. well i can remove the Debug impls.
the Clone is used to allow stacking providers -- the StsProvider creates an StsClient to do its work. Creating a client consumes the provider so I added the ability to clone the providers that rusoto provides to allow them all to be used with this system.